lib/ansible/inventory/manager.py

```diff
@@ -235,9 +238,9 @@ def parse_source(self, source, cache=False):
                     continue
 
                 # recursively deal with directory entries
-                b_fullpath = os.path.join(b_source, i)
-                parsed_this_one = self.parse_source(b_fullpath, cache=cache)
-                display.debug(u'parsed %s as %s' % (to_text(b_fullpath), parsed_this_one))
+                fullpath = to_text(os.path.join(b_source, i))
```

Comment:
This looks good but use an error handler so that we don't mangle non-utf8 characters on python2:
```
fullpath = to_text(os.path.join(b_source, i), errors='surrogate_or_strict')
```